pkg/driver/trace_driver.go

```diff
@@ -203,43 +211,51 @@ func (d *Driver) invokeFunction(metadata *InvocationMetadata) {
 	defer metadata.AnnounceDoneWG.Done()
 
 	var success bool
-
+	node := metadata.RootFunction.Front()
 	var record *mc.ExecutionRecord
-	switch d.Configuration.LoaderConfiguration.Platform {
-	case "Knative":
-		success, record = InvokeGRPC(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			d.Configuration.LoaderConfiguration,
-		)
-	case "OpenWhisk":
-		success, record = InvokeOpenWhisk(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			metadata.AnnounceDoneExe,
-			metadata.ReadOpenWhiskMetadata,
-		)
-	case "AWSLambda":
-		success, record = InvokeAWSLambda(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			metadata.AnnounceDoneExe,
-		)
-	case "Dirigent":
-		success, record = InvokeDirigent(
-			metadata.Function,
-			metadata.RuntimeSpecifications,
-			d.Configuration.LoaderConfiguration,
-		)
-	default:
-		log.Fatal("Unsupported platform.")
-	}
-
-	record.Phase = int(metadata.Phase)
-	record.InvocationID = composeInvocationID(d.Configuration.TraceGranularity, metadata.MinuteIndex, metadata.InvocationIndex)
-
-	metadata.RecordOutputChannel <- record
+	var runtimeSpecifications *common.RuntimeSpecification
+	for node != nil {
```

Comment:
Are you sure this loop is going to terminate? The list is double-linked and has no end.

Reply:
Yes. The list is not connected as a loop. The ```DAGCreation()``` function will add each function as the next of the previous function (next of function 0 will be function 1 ...). When ```node.Next()``` is called at the last function, it will return nil. Thus the condition ``` for node != nil ``` will be false and the for loop will terminate

Reply:
The ```DAGMode``` option is only used in ```internalRun()``` and takes in the list of functions to create the doubly linked-list. The logic of traversing through the list is done by ```invokeFunction()```. For the unit test, is it okay to simulate the initial part (creating the doubly linked list) and call ```invokeFunction()```  to emulate DAGMode being enabled? Such as in the most recent commit